R/luz.R

```diff
@@ -0,0 +1,64 @@
+#' @rdname vetiver_create_description
+#' @export
+vetiver_create_description.luz_module_fitted <- function(model) {
+    glue("A luz module")
+}
+
+#' @rdname vetiver_create_meta
+#' @export
+vetiver_create_meta.luz_module_fitted <- function(model, metadata) {
+    pkgs <- c("luz", model$model$required_pkgs)
+    vetiver_meta(metadata, required_pkgs = pkgs)
+}
+
+#' @rdname vetiver_create_ptype
+#' @export
+vetiver_ptype.luz_module_fitted <- function(model, ...) {
+    rlang::check_dots_used()
+    dots <- list(...)
+    check_ptype_data(dots)
+
+    data <- dots$prototype_data
+    data <- tensors_to_array(data)
+
+    # luz doesn't require named inputs. we use `input` if the user didn't provide any.
+    if (!is.list(data)) {
+        data <- tibble::tibble(input = data)
+    }
+
+    ptype <- vctrs::vec_ptype(data)
+    tibble::as_tibble(ptype)
+}
+
+#' @rdname vetiver_create_description
+#' @export
+vetiver_prepare_model.luz_module_fitted <- function(model) {
+    bundle::bundle(model)
+}
+
+#' @rdname handler_startup
+#' @export
+handler_startup.luz_module_fitted <- function(vetiver_model) {
+    attach_pkgs(vetiver_model$metadata$required_pkgs)
+}
+
+#' @rdname handler_startup
+#' @export
+handler_predict.luz_module_fitted <- function(vetiver_model, ...) {
+    force(vetiver_model)
```

Comment:
Can you tell me a little more about this?

Reply:
Since we return the closure directly, without evaluating `vetiver_model`, it will not be in the in the function env until the first closure call - but at this point `vetiver_model` could potentially have been garbage collected. This might not be the case for vetiver though, just feels like a good practice to `force` before returning the closure. 

I'm trying to avoid something like this:

``` r
f <- function(a, force) {
    if (force) force(a)
    function() {
        a + 1
    }
}

b <- 1
fun_f <- f(a = b, force = TRUE)
fun_nf <- f(a = b, force = FALSE)
rm(b);gc()
#>           used (Mb) gc trigger (Mb) limit (Mb) max used (Mb)
#> Ncells  674744 36.1    1413787 75.6         NA   710975 38.0
#> Vcells 1201226  9.2    8388608 64.0      32768  1888973 14.5

fun_f()
#> [1] 2
fun_nf()
#> Error in fun_nf(): object 'b' not found
```

Reply:
Thanks for the details!